Updated UI for Admin-> User Profile

pranshugupta54 opened the pull request:
**What kind of change does this PR introduce?**

Admin -> User Profile -> Re-design

**Issue Number:**

Fixes #1615 

Screenshot:
<img width="1439" alt="image" src="https://github.com/PalisadoesFoundation/talawa-admin/assets/76090263/6cc971a7-e49b-45e0-8322-98aeeeb988b1">


**Other information**
- The left side panel (according to Figma) should be fixed in another issue to match the design.
- The Top side panel (according to Figma) should be fixed in another issue to match the design.

**NOTE:**
- Some features are not yet supported by API

Cioppolo14 commented:
@pranshugupta54 Please fix the failing tests.

pranshugupta54 commented on src/screens/MemberDetail/MemberDetail.tsx:
```diff
@@ -124,195 +274,344 @@
+                        <img
+                          className={`rounded-circle mx-auto`}
+                          style={{ width: '80px', aspectRatio: '1/1' }}
+                          src={formState.file || userData?.user?.image}
```
Fixed

pranshugupta54 commented:
Hey @Cioppolo14, sorry for the delay. I've fixed the test for MemberDetail, all the 9 tests are passing for the file.

rishav-jha-mech commented on src/screens/MemberDetail/MemberDetail.tsx:
```diff
@@ -74,252 +226,363 @@ const MemberDetail: React.FC<MemberDetailProps> = ({ id }): JSX.Element => {
-      }
-    }
-  };
+  const sanitizedSrc = sanitizeHtml(formState.file, {
```
Are we previewing the uploaded image ?, is that why you have added this package here ?, Because sanitization of uploaded files is generally done on the server.

pranshugupta54 replied:
> Are we previewing the uploaded image ?, is that why you have added this package here ?, Because sanitization of uploaded files is generally done on the server.

Yes, I've made the formState in such a way to preview things like `firstName` and `avatar` on the right box
<img width="429" alt="image" src="https://github.com/PalisadoesFoundation/talawa-admin/assets/76090263/27d651a1-1760-4241-bfd6-9adc1d1f06dc">

rishav-jha-mech replied:
@pranshugupta54 so you are saying that on clicking on the Choose File button, the image that you have selected will be visible at Profile Details card ?

pranshugupta54 replied:
@rishav-jha-mech, yes. Also if you change `firstName` in the input box, it will show in the Profile Details card. 
Now if you click "Save", it saves otherwise it's lost

pranshugupta54 replied:
Example: https://www.loom.com/share/959a80f3602b4894bf8acc7bf5e59161?sid=025def3a-4b56-426d-a6d3-2de1c5d67f0e

pranshugupta54 commented:
Fixed lint issue, failed test

Cioppolo14 commented:
@pranshugupta54 Are you working on the test application check?

pranshugupta54 commented:
> @pranshugupta54 Are you working on the test application check?

@Cioppolo14, fixed those! 
<img width="299" alt="image" src="https://github.com/PalisadoesFoundation/talawa-admin/assets/76090263/e7d626a4-f4c7-415a-bb48-46d9b8936161">

rishav-jha-mech reviewed (commented):
@pranshugupta54 

Add the following 

- Date input for birthdate
- Placeholders for **all the inputs**
- Select input for Employment and Marital Status
- Choose file grey btn should be of our theme colour

pranshugupta54 commented:
Hey @rishav-jha-mech, 
- [x] Added date picker for birthDate
- [x] Placeholder for all inputs (didn't add for dropdown, they automatically have "select" text)
- [x] Added `DynamicDropDown` component which takes these labels and values for dropdown, instead of making multiple components
- [ ] We'll have to use some external package or completely new component for that, react-bootstrap isn't providing an option for color, though it fits the theme.
- [ ] Unable to figure how phone number works, it gives error `Value is not a valid phone number of the form +17895551234 (7-15 digits)`

Cioppolo14 commented:
@pranshugupta54  Please fix the test application check.

pranshugupta54 commented:
Hey @Cioppolo14, fixed the failing tests for MemberDetails 

<img width="619" alt="image" src="https://github.com/PalisadoesFoundation/talawa-admin/assets/76090263/2f8b92f5-d426-4e2a-8440-71b41957e803">

pranshugupta54 commented:
Hey @Cioppolo14 / @rishav-jha-mech, is there anything else needed?

Cioppolo14 commented:
@pranshugupta54 Please fix the code coverage to pass the checks.

palisadoes commented:
This is an update on the PR merging freeze:

1. In the next few hours we will be merging the `develop-userTypeFix` branch into the `develop` branch. 
1. The `develop-userTypeFix` branch was created to fix a long standing design flaw where Admins were Admins of all organizations, not specific ones.
1. The `userType` field has been removed from the User collection and it has been replaced by an `appUserProfileId` field.
    1. This field is `null` if the user isn’t registered to use the apps. This will help people to add users manually during the event checkin process, or if an Admin wants to manually add someone in the Admin dashboard.
    1. When not `null` the `AppUserProfileID` will reference a `AppUserProfile` collection with App related information such as the organizations for which a user may be an Admin.
    1. The updated schema can be found here https://github.com/PalisadoesFoundation/talawa-api/blob/develop-userTypeFix/schema.graphql
    1. This is the parent issue that we have been using to track progress:
        1. https://github.com/PalisadoesFoundation/talawa-api/issues/1965
1. This merge will cause some conflicts in your PR.

We decided to do this at the beginning of the weekend to give us all time to adjust PR code and create bug fixes that may arise.

Update your code at or after midnight GMT on the morning of March 23, 2024. (5:30am IST). 

If your PRs have already been approved, request a re-review after fixing the conflicts and refactoring to the new `AppUserProfileID` methodology.

pranshugupta54 commented:
Done for Dropdown too
<img width="1314" alt="image" src="https://github.com/PalisadoesFoundation/talawa-admin/assets/76090263/8f1a6c75-5de9-408d-a221-09287daf0681">

pranshugupta54 commented:
fixed lint issue

palisadoes commented:
## The PR merging freeze is lifted.

1. We are working on bug fixes that may arise. These should be few and are being tracked here:
    1. https://github.com/orgs/PalisadoesFoundation/projects/24
1. Please update your PRs and local repos. Fix any new merge conflicts that may have occurred.
    
Background:

1. The `develop-userTypeFix` branch was created to fix a long standing design flaw where Admins were Admins of all organizations, not specific ones.
1. The `userType` field has been removed from the User collection and it has been replaced by an `appUserProfileId` field.
    1. This field is `null` if the user isn’t registered to use the apps. This will help people to add users manually during the event checkin process, or if an Admin wants to manually add someone in the Admin dashboard.
    1. When not `null` the `AppUserProfileID` will reference a `AppUserProfile` collection with App related information such as the organizations for which a user may be an Admin.
    1. The updated schema can be found here https://github.com/PalisadoesFoundation/talawa-api/blob/develop-userTypeFix/schema.graphql
    1. This is the parent issue that we have been using to track progress:
        1. https://github.com/PalisadoesFoundation/talawa-api/issues/1965
1. If your PRs have already been approved, request a re-review after fixing the conflicts and refactoring to the new `AppUserProfileID` methodology.

Cioppolo14 commented:
@pranshugupta54 Please fix the failing tests.

pranshugupta54 commented:
Yes, fixing it completely for userType

pranshugupta54 commented:
@palisadoes, please check this.

Fixed everything,
- userType fix
- Testing done
- 100% code coverage
